lucene/core/src/java/org/apache/lucene/search/FieldValueHitQueue.java

```diff
@@ -121,7 +121,7 @@ protected boolean lessThan(final Entry hitA, final Entry hitB) {
   }
   
   // prevent instantiation and extension.
-  private FieldValueHitQueue(SortField[] fields, int size, boolean filterNonCompetitiveDocs) {
+  private FieldValueHitQueue(SortField[] fields, int size, boolean filterNonCompetitiveDocs, boolean hasAfter) {
```

Comment:
Not sure that `hasAfter` is really needed here.

Reply:
At this point of time, topValue for comparators is not set yet, that's why we need `hasAfter`.
As an alternative to this implementation, we can pass `FieldDoc after` to `FieldValueHitQueue.create` and setTopValue during `FieldValueHitQueue` creation.